### DISCUSSION TITLE
Context menu too wide

### DISCUSSION THREAD

**Discussion by bluee111** [tsujan/Kvantum#810] [General]
Good morning gentlemen, I don't know if you can help me with the desktop context menu, it is too wide, is it possible to modify it to be shorter like gnome or something close? I think I'm too blind that I didn't see it in Theme-Config.

![2022-07-22_20-27-min](https://user-images.githubusercontent.com/100516568/180586235-a077f277-0078-4581-93e1-bf9130576735.png)

**Comment by tsujan** [comment]
Good morning, dear user!

> it is too wide

It isn't. Look at the end of first label and the start of the last shortcut. Labels and shortcuts are in separate columns with a space in between.

**Reply by tsujan** [reply]
> shorter like gnome or something close?

Gnome's menus have even a wider space between columns:

![gnome](https://user-images.githubusercontent.com/981076/180587163-a5d7d5c6-3bfa-4182-943a-f6daef453845.png)

**Comment by bluee111** [comment]
so any way to get a result like this? 
![2022-07-22_21-37](https://user-images.githubusercontent.com/100516568/180587588-1899cbc4-0628-4d10-b9c7-ecd9fbf5295f.png)

**Reply by tsujan** [reply]
> so any way to get a result like this?

No. In this screenshot, the shortcut of the first item isn't clearly distinguished from its label. That's a problem Kvantum doesn't have.

Kvantum just puts a space equal to three times the font height between the label and shortcut. The rest is done by Qt itself, not by Kvantum.

Actually, Fusion's menus are wider than Kvantum's (Fusion is the default style of Qt).

**Comment by tsujan** [comment]
There's another factor too: the contents size that the app may give to the menu. The following Kvantum menu belongs to QTerminal. It has wider shortcuts but the menu is smaller:

![menu](https://user-images.githubusercontent.com/981076/180588623-505be4e0-1fb0-4e62-8808-f769cb5f5e9c.png)

**Comment by bluee111** [comment]
Now I understand, so removing the keyboard shortcuts in the desktop context menu is also handled by QT, right?
![2022-07-22_22-23](https://user-images.githubusercontent.com/100516568/180588843-1a491f46-2fd4-46a7-8986-29173c1bb054.png)

**Reply by tsujan** [reply]
> so removing the keyboard shortcuts in the desktop context menu is also handled by QT, right?

In a sense, yes. There is the key `QPlatformTheme::ShowShortcutsInContextMenus`. Its value is determined by the DE. Both KDE and LXQt set it to `true` because they want the shortcuts to be visible.

I don't know whether KDE has a setting for it. LXQt doesn't.

**Reply by tsujan** [reply]
This is what KDE did to prevent lack of shortcuts in menus: https://phabricator.kde.org/D8466?id=21293

LXQt did the same thing too.

**Reply by bluee111** [reply]
ok thanks for all the information :)
